rundeckapp/grails-app/services/rundeck/services/asyncimport/AsyncImportService.groovy

```diff
@@ -0,0 +1,1073 @@
+    static final String MODEL_PROJECT_NAME_EXT = '.jar'
+    static final String MODEL_PROJECT_INTERNAL_PREFIX = 'rundeck-'
+    static final String MAX_EXECS_PER_DIR_PROP_NAME = "asyncImportConfig.maxDistributedExecutions"
+    static final int KILOBYTE=1024
+
+    static final String EXECUTION_FILE_PREFIX = 'execution-'
+    static final String EXECUTION_FILE_EXT = '.xml'
+    static final String OUTPUT_FILE_PREFIX = 'output-'
+    static final String OUTPUT_FILE_EXT = '.rdlog'
+    static final String STATE_FILE_PREFIX = 'state-'
+    static final String STATE_FILE_EXT = '.state.json'
+
+    /**
+     *
+     * Creates the status file that will be the main report to inform about the whole process,
+     * a project cannot have more than a single status file.
+     *
+     * The file will be stored in project's file storage (Storage table in db)
+     *
+     * @param projectName - Required param to project.
+     *
+     * @return Boolean - "true" if the status file is created.
+     */
+    @Override
+    Boolean createStatusFile(String projectName) {
+        try {
+            if( !statusFileExists(projectName) ){
+                saveAsyncImportStatusForProject(projectName)
+                return true
+            }
+            return false
+        } catch (IOException e) {
+            e.printStackTrace();
+            throw e
+        }
+    }
+
+    /**
+     * True/false if status file exists in DB.
+     *
+     * @param projectName
+     * @return
+     */
+    Boolean statusFileExists(String projectName){
+        try {
+            def fwkProject = frameworkService.getFrameworkProject(projectName)
+            def statusFilepath = "${JSON_FILE_PREFIX}${projectName}${JSON_FILE_EXT}"
+            if( !fwkProject.existsFileResource(statusFilepath) ){
+                return false
+            }
+            return true
+        } catch (Exception e) {
+            e.printStackTrace();
+            throw e
+        }
+    }
+
+    /**
+     *
+     * Gets the status file content as an object.
+     *
+     * @param projectName - Required param to project.
+     *
+     * @return AsyncImportStatusDTO - DTO with all the status file content.
+     */
+    @Override
+    AsyncImportStatusDTO getAsyncImportStatusForProject(String projectName, ByteArrayOutputStream out = null) {
+        try{
+            if( out == null ){
+                out = new ByteArrayOutputStream()
+            }
+            final def fwkProject = frameworkService.getFrameworkProject(projectName)
+            fwkProject.loadFileResource(JSON_FILE_PREFIX + projectName + JSON_FILE_EXT, out)
+            def obj = new JsonSlurper().parseText(out.toString()) as AsyncImportStatusDTO
+            log.debug("Async Import status file content: ${obj.toString()}")
+            return obj
+        }catch(Exception e){
+            log.error("Error during the async import file extraction process: ${e.message}")
+            throw e
```

Comment:
shouldn't this throw a different kind of exception? (like AsyncImportException or so?)